### PR TITLE
Fix log_config

### PR DESCRIPTION
**Pull request by josejg**
Foundry's train and eval scripts use the `log_config` helper to log the parameter config. However, `log_config` breaks the logger abstraction by directly logging to wandb and/or mlflow.

This PR uses the `logger.log_hyperparameter` method which relies on the same method calls for wandb and mlflow. In addition this PR allows custom loggers to be able to log the config which is not currently supported.

EDIT: Run Name `210m-tpr100-adam-cool-linear-lr2d3-20pct-JvGeHi`

### CODE REVIEW

**Review by snarayan21 — COMMENTED** [review]
looks fine to me, given the changes around config logging would also like @b-chu to approve before merging.

Could you do a simple manual run off this branch to show the functionality working fine?

**Review by b-chu — COMMENTED** [review]
Breaks BC, but I'll let @snarayan21 approve. Do you have a screenshot of wandb and mlflow before and after this change?

File: llmfoundry/utils/config_utils.py
Comment (dakinggg) [reply llmfoundry/utils/config_utils.py:579]: yeah i think its fine to break BC here

**Review by snarayan21 — APPROVED** [review]
chilling

**Comment by b-chu** [comment]
Disabling automerge because I want to see how logging UI changed

> Do you have a screenshot of wandb and mlflow before and after this change?

**Comment by snarayan21** [comment]
@josejg merge whenever you're ready

**Review by snarayan21 — APPROVED** [review]
lgtm, just run name in description. Tests are failing though?

**Pull request merged**
